### PR TITLE
ROX-26699: Introduce unified transmissions metric

### PR DESCRIPTION
**Pull request by Maddosaurus**
### Description

This PR introduces updated metrics to better address the new reality of two scanner versions being available for generating Node Package information.
No metric is removed, as we need to keep the old metrics around for as long as Scanner v2 is officially supported to maintain backwards compatibility.
The idea is to introduce new versions of the metrics with correct names, deprecating the old ones where possible. For this case, metrics are duplicated and given the correct name and help text.
Some metrics are directly observed in the old Scanner v2 code (stackrox/scanner), . In these cases, the metric is created seperately and exists once for Scanner v2 and once for Scanner v4.

Overview: 
- numberofRHELPackages: **Duplication added**
- numberOfContentSets: **Duplication added**
- scanDuration / indexDuration: Two separated metrics (code called in Scanner v2)
- callToNodeInventoryDuration: Not applicable to v4
- rescanInterval: Updated help text
- protobufMessageSize: **Duplication added**
- scansTotal / indexesTotal: Two separated metrics (code called in Scanner v2)
- inventoryTransmissions: **Duplication added**

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

- Deploy ACS on OpenShift with `ROX_SCANNER_V4=true MONITORING_SUPPORT=true`
- Start debug pod on one of the collector pods: `kubectl -n stackrox debug -it collector-6255l --image=nicolaka/netshoot`
- curl compliance metrics: 
```
curl -s http://localhost:9091/metrics | grep -i package
# HELP rox_compliance_node_package_reports_total Number of total v2 node inventory scans and v4 index reports sent to sensor
# TYPE rox_compliance_node_package_reports_total counter
rox_compliance_node_package_reports_total{node_name="mm-10-24-nodescanning-lcfl7-worker-c-7jt5l",scanner_version="Scanner V4",transmission_type="scanning"} 3
rox_compliance_node_package_reports_total{node_name="mm-10-24-nodescanning-lcfl7-worker-c-7jt5l",scanner_version="Stackrox Scanner",transmission_type="scanning"} 3

```
Ensure this number is the same than its sister metric, `inventory_transmissions_total`: 
```
curl -s http://localhost:9091/metrics | grep -i inventory_transmissions_total
# HELP rox_compliance_inventory_transmissions_total Number of node inventory scans sent to sensor
# TYPE rox_compliance_inventory_transmissions_total counter
rox_compliance_inventory_transmissions_total{node_name="mm-10-24-nodescanning-lcfl7-worker-c-7jt5l",scanner_version="Scanner V4",transmission_type="scanning"} 3
rox_compliance_inventory_transmissions_total{node_name="mm-10-24-nodescanning-lcfl7-worker-c-7jt5l",scanner_version="Stackrox Scanner",transmission_type="scanning"} 3

```

Repeat these steps for the remaining added metrics

### CODE REVIEW

File: compliance/collection/metrics/metrics.go [hunk 1]
```diff
@@ -300,10 +315,20 @@ func ObserveNodeInventorySending(nodeName string, sendingType InventoryTransmiss
...
 }
 
+// ObserveNodePackageReportTransmissions observes the metric.
+func ObserveNodePackageReportTransmissions(nodeName string, sendingType InventoryTransmission, scannerVersion string) {
+	nodePackageReportTransmissions.With(prometheus.Labels{
+		"node_name":         nodeName,
+		"transmission_type": string(sendingType),
+		"scanner_version":   scannerVersion,
+	}).Inc()
+}
```
Comment (vikin91) [inline compliance/collection/metrics/metrics.go:325]: How about having single observe function that updates both metrics?

```suggestion
func ObserveNodePackageReportTransmissions(nodeName string, sendingType InventoryTransmission, scannerVersion string) {	
	nodePackageReportTransmissions.With(prometheus.Labels{
		"node_name":         nodeName,
		"transmission_type": string(sendingType),
		"scanner_version":   scannerVersion,
	}).Inc()
	// Recorded additionally in the old metric for backwards compatibility.
	// Remove the block below after dropping Stackrox Scanner (aka Scanner v2).
	inventoryTransmissions.With(prometheus.Labels{
		"node_name":         nodeName,
		"transmission_type": string(sendingType),
		"scanner_version":   scannerVersion,
	}).Inc()
}
```

**Review by vikin91 — APPROVED** [review]
Looks good! I have one idea for improvement - see comment.

There are many metrics that are affected by this issue in `compliance/collection/metrics/metrics.go`. Do you plan to open PRs for the rest of them?

**Comment by Maddosaurus** [comment]
> There are many metrics that are affected by this issue

That's a good point. I've carefully checked the remaining metrics. Let me give an overview:
- numberofRHELPackages: **Duplication added**
- numberOfContentSets: **Duplication added**
- scanDuration / indexDuration: Two separated metrics (code called in Scanner v2)
- callToNodeInventoryDuration: Not applicable to v4
- rescanInterval: Updated help text
- protobufMessageSize: **Duplication added**
- scansTotal / indexesTotal: Two separated metrics (code called in Scanner v2)
- inventoryTransmissions: Duplication added (see initial commit)

**Comment by vikin91** [comment]
> > There are many metrics that are affected by this issue
> 
> That's a good point. I've carefully checked the remaining metrics. Let me give an overview:
> 
> * numberofRHELPackages: **Duplication added**
> * numberOfContentSets: **Duplication added**
> * scanDuration / indexDuration: Two separated metrics (code called in Scanner v2)
> * callToNodeInventoryDuration: Not applicable to v4
> * rescanInterval: Updated help text
> * protobufMessageSize: **Duplication added**
> * scansTotal / indexesTotal: Two separated metrics (code called in Scanner v2)
> * inventoryTransmissions: Duplication added (see initial commit)

That is a very nice overview and it made it easier to review the code changes. Many thanks for adding it!

**Review by vikin91 — APPROVED** [review]
The new iteration looks great! Thanks for improving!

**Pull request merged**